docs/changing-the-ast.md

```diff
@@ -0,0 +1,66 @@
+---
+title: Changing the AST
+category: Compiler Internals
+categoryindex: 200
+index: 800
+---
+# Changing the AST
+
+Making changes to the AST is a common task when working on new F# compiler features or when working on developer tooling.  
+This document describes the process of making changes to the AST.
+
+The easiest way to modify the AST is to start with the type definitions in `SyntaxTree.fsi` and `SyntaxTree.fs` and then let the compiler guide you to the places where you need to make changes.
+Let's look at an example. Here, we want to add a new field `divRange` to the union case `Rational` of `SynRationalConst`:
+
+```fsharp
+type SynRationalConst =
+
+    // ...
+
+    | Rational of
+        numerator: int32 *
+        numeratorRange: range *
+        divRange: range *   // our new field
+        denominator: int32 *
+        denominatorRange: range *
+        range: range
+    
+    // ...
+```	
+
+After modifying `SyntaxTree.fsi` and `SyntaxTree.fs`, the compiler will report erros in `pars.fsy`.
```

Comment:
Doing a build of `FSharp.Compiler.Service.fsproj` doesn't always trigger lexing and yaccing for me. Maybe mentioned that a `rebuild` will always do these things.

Reply:
`<unrelated-to-this-PR>`
When doing frequent changes, I set the the pars.fsy file to be "Copy if newer", that way change-detection works.
But is something I only changed locally for myself on my machine, since it also leads to including the file in the output folder... (wonder if there is a better way)

`</unrelated-to-this-PR>`